src/NServiceBus.Testing/Sagas/TestableSaga.cs

```diff
@@ -213,6 +213,7 @@ async Task<HandleResult> InnerHandle(QueuedSagaMessage message, string handleMet
             using (var session = new NonDurableSynchronizedStorageSession())
             {
                 var contextBag = new ContextBag();
```

Comment:
I'm wondering if it would be better to use `context.Extensions` from the `TestableHandlerContext` instead of newing a context bag to simulate more how the persister would behave, since it would actually pass the same context bag to the methods or did we deliberately not do this to avoid any side effects?

Reply:
Agree, will change. Assume it will pass tests.